addons/account/models/account_payment.py

```diff
@@ -539,6 +539,9 @@ def action_validate_invoice_payment(self):
         if any(len(record.invoice_ids) != 1 for record in self):
             # For multiple invoices, there is account.register.payments wizard
             raise UserError(_("This method should only be called to process a single invoice's payment."))
+
+        # Make sure that only the commercial partner is registered for payment
+        self.partner_id = self.partner_id.commercial_partner_id
```

Comment:
Ooch no, we can't change such a value in the back of the user just before posting.